core/arns/src/main/java/software/amazon/awssdk/arns/ArnResource.java

```diff
@@ -103,22 +113,27 @@ public static ArnResource fromString(String resource) {
         int resourceColonIndex = resource.indexOf(splitter, resourceTypeColonIndex);
         int qualifierColonIndex = resource.indexOf(splitter, resourceColonIndex + 1);
         if (qualifierColonIndex < 0) {
-            builder.resource(resource.substring(resourceTypeColonIndex + 1));
+            builder.resourceId(resource.substring(resourceTypeColonIndex + 1));
         } else {
-            builder.resource(resource.substring(resourceTypeColonIndex + 1, qualifierColonIndex));
+            builder.resourceId(resource.substring(resourceTypeColonIndex + 1, qualifierColonIndex));
             builder.qualifier(resource.substring(qualifierColonIndex + 1));
         }
+        if (splitter.equals('/')) {
+            builder.usingSlashSeperator();
+        }
 
         return builder.build();
     }
 
     @Override
     public String toString() {
```

Comment:
This may potentially break customers who rely on existing behavior. Can we introduce a new method, `toStringPretty` or `toStringFormatted`?